media/js/app/taxonomy/taxonomy.js

```diff
@@ -612,10 +546,48 @@
                             });
                         }
                     }
+                    //once we get to last iteration we save it all
+                     if (i == MAX - 1) {
```

Comment:
Wow thanks for that tip, I didn't know variables were in the same scope as the for loop they're declared in for javascript.